frontend/public/components/edit-yaml.jsx

```diff
@@ -261,11 +257,66 @@ export const EditYAML = connect(stateToProps)(
         }
       }
 
+      return yaml;
+    }
+
+    loadYaml(reload = false, obj = this.props.obj) {
+      if (this.state.initialized && !reload) {
+        return;
+      }
+
+      const yaml = this.convertObjToYAMLString(obj);
+
       this.displayedVersion = _.get(obj, 'metadata.resourceVersion');
       this.setState({ yaml, initialized: true, stale: false });
       this.resize();
     }
 
+    addToYAML(obj) {
+      const yaml = this.convertObjToYAMLString(obj);
+
+      const selection = this.monacoRef.current.editor.getSelection();
+      const range = new window.monaco.Range(
+        selection.startLineNumber,
+        selection.startColumn,
+        selection.endLineNumber,
+        selection.endColumn,
+      );
+
+      // Grab the current position and indent every row but the first row to that indentation
+      const gap = new Array(selection.startColumn).join(' ');
+      const lines = yaml.split('\n');
+      const lineCount = lines.length;
+      const indentedText = lines
+        .map((line, i) => {
+          if (i === 0) {
+            // Already indented, leave it alone
+            return line;
+          }
+          return `${gap}${line}`;
+        })
+        .join('\n');
+
+      const ident = { major: 1, minor: 1 };
+      const op = { identifier: ident, range, text: indentedText, forceMoveMarkers: true };
+      this.monacoRef.current.editor.executeEdits('some-source-id', [op]);
```

Comment:
I forgot to change this snippet. `some-source-id` was not a good name lol.

Reply:
Reworked the insert and cleaned up some stuff / improved the comments so it's more clear what I am doing and why.